MapboxDirections/MBVisualInstruction.swift

```diff
@@ -2,127 +2,99 @@ import Foundation
      */
-    @objc public let secondaryText: String?
+    @objc public var maneuverType: ManeuverType
     
     /**
      :nodoc:
-     Ancillary visual information about the `RouteStep`.
-     
-     This is the structured representation of `secondaryText`.
+     The modifier type for the `VisualInstruction`.
      */
-    @objc public let secondaryTextComponents: [VisualInstructionComponent]?
-    
+    @objc public var maneuverDirection: ManeuverDirection
     
     /**
      :nodoc:
-     Indicates what side of a bidirectional road the driver must be driving on. Also referred to as the rule of the road.
+     Most important visual content to convey to the user about the `RouteStep`.
      */
-    @objc public var drivingSide: DrivingSide
+    @objc public let textComponents: [VisualInstructionComponent]
     
     /**
-     :nodoc:
-     Initialize a `VisualInstruction` from a dictionary given a `DrivingSide`.
+     The degrees at which you will be exiting a roundabout, assuming 180 indicates going straight through the roundabout.
      */
-    @objc public convenience init(json: [String: Any], drivingSide: DrivingSide) {
-        let distanceAlongStep = json["distanceAlongGeometry"] as! CLLocationDistance
-        
-        let primaryTextComponent = json["primary"] as! JSONDictionary
-        let primaryText = primaryTextComponent["text"] as! String
-        let primaryManeuverType = ManeuverType(description: primaryTextComponent["type"] as! String) ?? .none
-        let primaryManeuverDirection = ManeuverDirection(description: primaryTextComponent["modifier"] as! String)  ?? .none
-        let primaryTextComponents = (primaryTextComponent["components"] as! [JSONDictionary]).map {
-            VisualInstructionComponent(maneuverType: primaryManeuverType, maneuverDirection: primaryManeuverDirection, json: $0)
-        }
-        
-        var secondaryText: String?
-        var secondaryTextComponents: [VisualInstructionComponent]?
-        if let secondaryTextComponent = json["secondary"] as? JSONDictionary {
-            secondaryText = secondaryTextComponent["text"] as? String
-            let secondaryManeuverType = ManeuverType(description: secondaryTextComponent["type"] as! String) ?? .none
-            let secondaryManeuverDirection = ManeuverDirection(description: secondaryTextComponent["modifier"] as! String)  ?? .none
-            secondaryTextComponents = (secondaryTextComponent["components"] as! [JSONDictionary]).map {
-                VisualInstructionComponent(maneuverType: secondaryManeuverType, maneuverDirection: secondaryManeuverDirection, json: $0)
-            }
-        }
-        
-        self.init(distanceAlongStep: distanceAlongStep, primaryText: primaryText, primaryTextComponents: primaryTextComponents, secondaryText: secondaryText, secondaryTextComponents: secondaryTextComponents, drivingSide: drivingSide)
-    }
+    @objc public var degrees: CLLocationDegrees = 180
```

Comment:
Why not make this optional and nil by default?

Reply:
See https://github.com/mapbox/MapboxDirections.swift/pull/259#discussion_r180805941